flexget/plugins/sites/lostfilm.py

```diff
@@ -47,151 +38,115 @@ class LostFilm(object):
+            for seen_btn in seen_buttons:
                 try:
-                    torrent_link = item.find(
-                        'div', 'inner-box--link sub').a['href']
-                    quality = item.find('div', 'inner-box--label').text.strip()
-                except:
-                    log.debug('Item doesn\'t have a link or quality')
-                    continue
-                if torrent_link is None or quality is None:
-                    log.debug('Item doesn\'t have a link or quality')
+                    details_pane = seen_btn.parent.find('div', 'details-pane')
+                    episode_name_rus = details_pane.find('div', 'alpha').text.strip()
+                    episode_name_eng = details_pane.find('div', 'beta').text.strip()
+                    series_name_rus = details_pane.parent.find('div', 'name-ru').text.strip()
+                    series_name_eng = details_pane.parent.find('div', 'name-en').text.strip()
+                    lostfilm_num, season_num, episode_num = [int(x) for x in seen_btn['data-code'].split('-')]
+                except AttributeError:
+                    log.exception('Could not parse %s, probably HTML markup was changed', release_url)
                     continue
-                if quality_map.get(quality):
-                    quality, file_ext = quality_map.get(quality)
-                else:
-                    file_ext = 'avi'
-                if series_name_eng:
-                    new_title = '.'.join(
-                        [
-                            series_name_eng,
-                            episode_id,
-                            quality,
-                            'rus.LostFilm.TV',
-                            file_ext,
-                            'torrent'
-                        ]
-                    ).replace(' ', '.')
-                else:
-                    if item.get('title') is not None:
-                        new_title = '{} {}'.format(
-                            item['title'],
-                            quality)
-                    else:
-                        log.debug('Item doesn\'t have a title')
-                        continue
-                new_entry = Entry()
-                new_entry['url'] = torrent_link
-                new_entry['title'] = new_title.strip()
-                if series_name_rus:
+
+
+                torrents = self.get_torrents(task, lostfilm_num, season_num, episode_num)
+                for quality, torrent_link in torrents.items():
+                    new_entry = Entry()
+                    new_entry['title'] = '{:s} S{:02d}E{:02d} - {:s} [{:s}]'.format(series_name_eng, season_num, episode_num, episode_name_eng, quality)
+                    new_entry['url'] = torrent_link
+                    new_entry['name'] = series_name_eng
+                    new_entry['season'] = season_num
+                    new_entry['episode'] = episode_num
+                    new_entry['quality'] = quality
```

Comment:
You shouldn't set the quality field. Having it in the title is enough.

Reply:
Removed